pkg/metrics/prometheus_exporter.go

```diff
@@ -0,0 +1,67 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	"contrib.go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/stats/view"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	curPromSrv    *http.Server
+	curPromSrvMux sync.Mutex
+)
+
+var log = logf.Log.WithName("metrics")
+
+const namespace = "gatekeeper"
+
+func newPrometheusExporter() (view.Exporter, error) {
+	e, err := prometheus.NewExporter(prometheus.Options{Namespace: namespace})
+	if err != nil {
+		log.Error(err, "Failed to create the Prometheus exporter.")
+		return nil, err
+	}
+	log.Info("Starting server for OpenCensus Prometheus exporter")
+	// Start the server for Prometheus scraping
+	go func() {
+		srv := startNewPromSrv(e, *prometheusPort)
+		if err := srv.ListenAndServe(); err != nil {
+			log.Error(err, "Failed to ListenAndServe")
```

Comment:
should we bubble up this error?

Reply:
this is inside a goroutine, so it might be a little tricky to bubble that error

Reply:
If the Prometheus exporter were a Runnable that could be fed to the manager, that would allow failure of the Prometheus server to cause the exit of the program.

Runnables are meant to block until failure, at which point the manager shuts down.

Reply:
@maxsmythe updated to return back to manager